docs/testing.md

```diff
@@ -83,4 +83,10 @@ Unit tests are meant to extensively test the functionality of a HubTurbo compone
 
 ## Additional Tools
 
-- [mockito](http://mockito.org/) is used in HubTurbo's tests suite to create clean and verifiable stubbed classes. A common use case is to isolate a small component under test while mocking other required components without having to write complex stubbing logic and mixing between application and testing code. Refer to usage of *mockito* in our tests suite and [mockito's docs](http://mockito.github.io/mockito/docs/current/org/mockito/Mockito.html) for more information, guides and caveats when using *mockito*.
+- [mockito](http://mockito.org/) is used in HubTurbo's tests suite to create clean and verifiable stubbed classes.
+A common use case is to isolate a small component under test while mocking other required components without having to write complex stubbing logic and mixing between application and testing code. Refer to usage of *mockito* in our tests suite and [mockito's docs](http://mockito.github.io/mockito/docs/current/org/mockito/Mockito.html) for more information, guides and caveats when using *mockito*.
+- [MockServer](http://www.mock-server.com) is used in HubTurbo' tests suite to emulate and verify interactions with GitHub API services.
+Note that when passing in `hostname` other than `api.github.com` (usually `"localhost"` when MockServer is used) to `egit`'s `GitHubClient` constructor, subsequent request paths will be appended to the `/api/v3` prefix.
+Also note that if you receive a 404 for a valid request path and parameters, a possible cause is that the MockServer process may take while to build the expectations when a large amount of request or response data is involved.
```

Comment:
This only happens if we only get the headers from multiple requests (HEAD requests), which take very short time, while the body is still building up. If we request for the body (GET, POST etc.) then it will wait until the body is complete before we get a response (so no problems). So this can be completely avoided if we don't put body data into requests that we only care about headers. Otherwise, just make a full request then it will block until everything is available.